tests/test.py

```diff
@@ -521,6 +522,14 @@ def test_export(self):
 
         self.assertEqual(exported_values, value_list)
 
+    def test_get_gid(self):
+        self.sheet._id = 'od6'
+        self.assertEqual(self.sheet.gid, '0')
+        self.sheet._id = 'osyqnsz'
+        self.assertEqual(self.sheet.gid, '1751403737')
+        self.sheet._id = 'ogsrar0'
```

Comment:
I'm getting  '400: Invalid query argument for grid_id.' error when I run the tests:
~~~
Traceback (most recent call last):
  File "~/b/p/gspread/tests/test.py", line 203, in tearDown
    self.spreadsheet.del_worksheet(self.sheet)
  File "~/b/p/gspread/gspread/models.py", line 156, in del_worksheet
    self.client.del_worksheet(worksheet)
  File "~/b/p/gspread/gspread/client.py", line 203, in del_worksheet
    self.session.delete(url)
  File "~/b/p/gspread/gspread/httpsession.py", line 76, in delete
    return self.request('DELETE', url, params=params, **kwargs)
  File "~/b/p/gspread/gspread/httpsession.py", line 69, in request
    response.status_code, response.content))
RequestError: (400, '400: Invalid query argument for grid_id.')
~~~

My guess is that this happens because of setting `self.sheet._id` to a value different from what it was before the test started. It could work for you since you're using a different document for testing. I see two possible ways on how to make the test document independent:

1. to extract the gid calculation function and move it to [utils](https://github.com/burnash/gspread/blob/4ad90b4523214b14e61ef889098603fa5d449c49/gspread/utils.py) module (since the only value the method is accessing now is `_id`). Since this function will be pure it can be tested without a spreadsheet document, as [other utils functions](https://github.com/burnash/gspread/blob/4ad90b4523214b14e61ef889098603fa5d449c49/tests/test.py#L51). 

2. to store `self.sheet._id` value before the test code and reset the value after. But that's an inferior approach, imho.